client/src/views/asset-list.js

```diff
@@ -1,12 +1,37 @@
 import Snabbdom from 'snabbdom-pragma'
-
+import { formatAssetAmount } from './util'
 import layout from './layout'
 
-export default ({ assetMap, t, ...S }) => {
+const staticRoot = process.env.STATIC_ROOT || ''
+export default ({ assetMap, assetList, loading, t, ...S }) => {
+
+  const { assets, query, total } = !assetList ? {} : assetList
```

Comment:
If the `assetList` isn't loaded yet, its better to avoid rendering the page entirely, rather than flashing the `No registered assets` message for a moment and then disappearing it when the assets finish loading.

A simple `if (!assetList) return;` should do the trick.

Reply:
When I added this line .... if (!assetList) return; The "loading spinner" wasn't showing.. 
When you navigate to the page the first time, it takes a few second for the AssetList to load. ..

const { assets, total } = !assetList ? {} : assetList
!assets ? <div className="load-more-container"><img src="img/Loading.gif" /></div>

Reply:
Yes, okay, makes sense. I see that you changed the code to not display the "No registered assets" message at all, but you could show it when `assets` is loaded but empty.

Reply:
Yes, okay, makes sense. I see that you changed the code to not display the "No registered assets" message at all, but you could show it when `assets` is loaded and is an empty list.

Reply:
changes have been made